spec/integration/hanami/router/body_parsing_middelware_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'hanami/routing/middleware/body_parser'
+
+RSpec.describe 'Body parsing' do
+  before do
+    endpoint = lambda { |env|
+      [200, {}, [env['router.params'].inspect]]
+    }
+
+    @routes = Hanami::Router.new do
+      patch '/books/:id',   to: endpoint
+      patch '/authors/:id', to: endpoint
+    end
+
+    hanami_app = Hanami::Routing::Middleware::BodyParser.new(@routes) do
+      add_parser :json
+      add_parser XmlMiddelwareParser
+    end
+
+    @app = Rack::MockRequest.new(hanami_app)
+  end
+
+  it 'is successful (JSON)' do
```

Comment:
what do you think to create a context for JSON and XML?

Reply:
Sure no problem